Flex depended on to build, but not listed as build dependency in debian/control, nor checked in ./configure

flex is required but not depended on in debian build dependencies, nor is it's presence checked for in configure script.

```
.....snip......
make[5]: Entering directory '/home/zaf/dev/amplet2/src/tests/http'
Making all in .
make[6]: Entering directory '/home/zaf/dev/amplet2/src/tests/http'
  CC       http.lo
  CC       servers.lo
  CC       parsers.lo
  CC       output.lo
  LEX      lexer.c
/home/zaf/dev/amplet2/missing: line 81: flex: command not found
WARNING: 'flex' is missing on your system.
         You should only need it if you modified a '.l' file.
         You may want to install the Fast Lexical Analyzer package:
         <http://flex.sourceforge.net/>
Makefile:582: recipe for target 'lexer.c' failed
make[6]: *** [lexer.c] Error 127
make[6]: Leaving directory '/home/zaf/dev/amplet2/src/tests/http'
```

Will submit pull request for this shortly

--- a/configure.ac
+++ b/configure.ac
@@ -16,6 +16,12 @@ AC_PROG_CC
 AM_PROG_CC_C_O
 AC_PROG_LIBTOOL
 
+# Checks for flex
+AC_PROG_LEX
+if test "$LEX" != "flex"; then
+        AC_MSG_ERROR(Required program flex not found)
+fi
+
 # Checks for libraries.
 AC_SEARCH_LIBS(dlopen, dl)
 

--- a/debian/control
+++ b/debian/control
@@ -2,7 +2,7 @@ Source: amplet2
 Section: net
 Priority: optional
 Maintainer: Brendon Jones <brendonj@waikato.ac.nz>
-Build-Depends: debhelper (>= 7.0.50~), autotools-dev, python, libunbound-dev, libssl-dev, libpcap-dev (>= 1.7.4), libyaml-dev, libprotobuf-c-dev, protobuf-c-compiler, protobuf-compiler, dh-systemd, libconfuse-dev, libcurl4-openssl-dev, librabbitmq-dev (>= 0.7.1), libwandevent-dev, python-setuptools
+Build-Depends: debhelper (>= 7.0.50~), autotools-dev, python, libunbound-dev, libssl-dev, libpcap-dev (>= 1.7.4), libyaml-dev, libprotobuf-c-dev, protobuf-c-compiler, protobuf-compiler, dh-systemd, libconfuse-dev, libcurl4-openssl-dev, librabbitmq-dev (>= 0.7.1), libwandevent-dev, python-setuptools, flex
 Standards-Version: 3.8.4
 Homepage: http://amp.wand.net.nz
 #Vcs-Git: git://git.debian.org/collab-maint/amplet2.git

